src/main/java/course_project/SeaBattle.java

```diff
@@ -0,0 +1,89 @@
+
+package course_project;
+
+import course_project.models.Ship;
+import course_project.service.ComputerThread;
+import course_project.service.ShipCreator;
+import java.util.*;
+import java.util.concurrent.Exchanger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import course_project.service.UserThread;
+
+public class SeaBattle {
```

Comment:
Quite difficult to play as I don't see the field, not mine, not opponents'. Also I can hit the same cell twice without errors

Reply:
Nice idea and concurrent implementation of computer/user threads.

Reply:
Pretty good abstracts separation.

Reply:
Well done! Good skills using Java Core and Java libraries. Nice abstractions segregation. Nice idea and concurrent implementation of computer/user threads. 

What to improve: 
Bug:I can hit the same cell twice without errors
Quite difficult to play as I don't see the field, not mine, not opponents'.
Also, usually concurrency is not used due to its complexity. So it can cause some unexpected issues in your application, especially because it will be really difficult to trace the cause. For learning purposes it's great, but I wouldn't use in production if not required